Maple2.Model/Game/FieldInstance.cs

```diff
@@ -5,14 +5,16 @@
 namespace Maple2.Model.Game;
 
 public sealed class FieldInstance : IByteSerializable {
-    public static FieldInstance Default = new(InstanceType.none, 0);
+    public static FieldInstance Default = new(InstanceType.none, 0, false);
 
     public readonly InstanceType InstanceType;
     public readonly int InstanceId;
+    public readonly bool SaveMap;
 
-    public FieldInstance(InstanceType instanceType, int instanceId) {
+    public FieldInstance(InstanceType instanceType, int instanceId, bool saveMap) {
```

Comment:
should we just use this as a metadata at this point instead of instantiating a new class?